newIDE/app/src/EventsSheet/ParameterFields/ObjectField.js

```diff
@@ -14,6 +14,41 @@ import { Trans } from '@lingui/macro';
 import { nameAndIconContainer } from '../EventsTree/ClassNames';
 import InAppTutorialContext from '../../InAppTutorial/InAppTutorialContext';
 
+const gd: libGDevelop = global.gd;
+
+export const getBehaviorConstraints = (
+  platform: gdPlatform,
+  functionMetadata: gdInstructionMetadata | gdExpressionMetadata,
+  parameterIndex: number
+) => {
+  const behaviorConstraints: Array<{
+    behaviorName: string,
+    behaviorType: string,
+  }> = [];
+  for (
+    let index = parameterIndex + 1;
+    index < functionMetadata.getParametersCount();
+    index++
+  ) {
+    const behaviorParameter = functionMetadata.getParameter(index);
+    if (behaviorParameter.getType() !== 'behavior') {
+      break;
+    }
+    const behaviorType = behaviorParameter.getExtraInfo();
+    const behaviorMetadata = gd.MetadataProvider.getBehaviorMetadata(
+      platform,
+      behaviorType
+    );
+    if (behaviorMetadata.isHidden()) {
```

Comment:
I don't understand why the only criterium to be a constraint is to be hidden

Reply:
Users can't add an hidden behavior so the instruction will never work for these objects.
For visible behaviors, the behavior field suggests to add the behavior to the object.
I'll add a comment.